// hume/dist/cjs/serialization/resources/empathicVoice/resources/configs/client/requests/PostedConfigVersion.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostedConfigVersion = void 0;
const core = __importStar(require("../../../../../../../core/index.js"));
const PostedBuiltinTool_js_1 = require("../../../../types/PostedBuiltinTool.js");
const PostedConfigPromptSpec_js_1 = require("../../../../types/PostedConfigPromptSpec.js");
const PostedEllmModel_js_1 = require("../../../../types/PostedEllmModel.js");
const PostedEventMessageSpecs_js_1 = require("../../../../types/PostedEventMessageSpecs.js");
const PostedLanguageModel_js_1 = require("../../../../types/PostedLanguageModel.js");
const PostedNudgeSpec_js_1 = require("../../../../types/PostedNudgeSpec.js");
const PostedTimeoutSpecs_js_1 = require("../../../../types/PostedTimeoutSpecs.js");
const PostedUserDefinedToolSpec_js_1 = require("../../../../types/PostedUserDefinedToolSpec.js");
const PostedWebhookSpec_js_1 = require("../../../../types/PostedWebhookSpec.js");
const VoiceRef_js_1 = require("../../../../types/VoiceRef.js");
exports.PostedConfigVersion = core.serialization.object({
    builtinTools: core.serialization.property("builtin_tools", core.serialization.list(PostedBuiltinTool_js_1.PostedBuiltinTool.optional()).optional()),
    ellmModel: core.serialization.property("ellm_model", PostedEllmModel_js_1.PostedEllmModel.optional()),
    eventMessages: core.serialization.property("event_messages", PostedEventMessageSpecs_js_1.PostedEventMessageSpecs.optional()),
    eviVersion: core.serialization.property("evi_version", core.serialization.string()),
    languageModel: core.serialization.property("language_model", PostedLanguageModel_js_1.PostedLanguageModel.optional()),
    nudges: PostedNudgeSpec_js_1.PostedNudgeSpec.optional(),
    prompt: PostedConfigPromptSpec_js_1.PostedConfigPromptSpec.optional(),
    timeouts: PostedTimeoutSpecs_js_1.PostedTimeoutSpecs.optional(),
    tools: core.serialization.list(PostedUserDefinedToolSpec_js_1.PostedUserDefinedToolSpec.optional()).optional(),
    versionDescription: core.serialization.property("version_description", core.serialization.string().optional()),
    voice: VoiceRef_js_1.VoiceRef.optional(),
    webhooks: core.serialization.list(PostedWebhookSpec_js_1.PostedWebhookSpec.optional()).optional(),
});
