// hume/dist/cjs/api/resources/empathicVoice/client/Client.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmpathicVoiceClient = void 0;
const BaseClient_js_1 = require("../../../../BaseClient.js");
const Client_js_1 = require("../resources/chat/client/Client.js");
const Client_js_2 = require("../resources/chatGroups/client/Client.js");
const Client_js_3 = require("../resources/chats/client/Client.js");
const Client_js_4 = require("../resources/configs/client/Client.js");
const Client_js_5 = require("../resources/controlPlane/client/Client.js");
const Client_js_6 = require("../resources/prompts/client/Client.js");
const Client_js_7 = require("../resources/tools/client/Client.js");
class EmpathicVoiceClient {
    constructor(options = {}) {
        this._options = (0, BaseClient_js_1.normalizeClientOptionsWithAuth)(options);
    }
    get controlPlane() {
        var _a;
        return ((_a = this._controlPlane) !== null && _a !== void 0 ? _a : (this._controlPlane = new Client_js_5.ControlPlaneClient(this._options)));
    }
    get chatGroups() {
        var _a;
        return ((_a = this._chatGroups) !== null && _a !== void 0 ? _a : (this._chatGroups = new Client_js_2.ChatGroupsClient(this._options)));
    }
    get chats() {
        var _a;
        return ((_a = this._chats) !== null && _a !== void 0 ? _a : (this._chats = new Client_js_3.ChatsClient(this._options)));
    }
    get configs() {
        var _a;
        return ((_a = this._configs) !== null && _a !== void 0 ? _a : (this._configs = new Client_js_4.ConfigsClient(this._options)));
    }
    get prompts() {
        var _a;
        return ((_a = this._prompts) !== null && _a !== void 0 ? _a : (this._prompts = new Client_js_6.PromptsClient(this._options)));
    }
    get tools() {
        var _a;
        return ((_a = this._tools) !== null && _a !== void 0 ? _a : (this._tools = new Client_js_7.ToolsClient(this._options)));
    }
    get chat() {
        var _a;
        return ((_a = this._chat) !== null && _a !== void 0 ? _a : (this._chat = new Client_js_1.ChatClient(this._options)));
    }
}
exports.EmpathicVoiceClient = EmpathicVoiceClient;
