// hume/dist/esm/api/resources/empathicVoice/client/Client.mjs
// This file was auto-generated by Fern from our API Definition.
import { normalizeClientOptionsWithAuth } from "../../../../BaseClient.mjs";
import { ChatClient } from "../resources/chat/client/Client.mjs";
import { ChatGroupsClient } from "../resources/chatGroups/client/Client.mjs";
import { ChatsClient } from "../resources/chats/client/Client.mjs";
import { ConfigsClient } from "../resources/configs/client/Client.mjs";
import { ControlPlaneClient } from "../resources/controlPlane/client/Client.mjs";
import { PromptsClient } from "../resources/prompts/client/Client.mjs";
import { ToolsClient } from "../resources/tools/client/Client.mjs";
export class EmpathicVoiceClient {
    constructor(options = {}) {
        this._options = normalizeClientOptionsWithAuth(options);
    }
    get controlPlane() {
        var _a;
        return ((_a = this._controlPlane) !== null && _a !== void 0 ? _a : (this._controlPlane = new ControlPlaneClient(this._options)));
    }
    get chatGroups() {
        var _a;
        return ((_a = this._chatGroups) !== null && _a !== void 0 ? _a : (this._chatGroups = new ChatGroupsClient(this._options)));
    }
    get chats() {
        var _a;
        return ((_a = this._chats) !== null && _a !== void 0 ? _a : (this._chats = new ChatsClient(this._options)));
    }
    get configs() {
        var _a;
        return ((_a = this._configs) !== null && _a !== void 0 ? _a : (this._configs = new ConfigsClient(this._options)));
    }
    get prompts() {
        var _a;
        return ((_a = this._prompts) !== null && _a !== void 0 ? _a : (this._prompts = new PromptsClient(this._options)));
    }
    get tools() {
        var _a;
        return ((_a = this._tools) !== null && _a !== void 0 ? _a : (this._tools = new ToolsClient(this._options)));
    }
    get chat() {
        var _a;
        return ((_a = this._chat) !== null && _a !== void 0 ? _a : (this._chat = new ChatClient(this._options)));
    }
}
