// hume/dist/cjs/serialization/resources/expressionMeasurement/resources/stream/resources/stream/types/StreamModelPredictions.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamModelPredictions = void 0;
const core = __importStar(require("../../../../../../../../core/index.js"));
const StreamModelPredictionsBurst_js_1 = require("./StreamModelPredictionsBurst.js");
const StreamModelPredictionsFace_js_1 = require("./StreamModelPredictionsFace.js");
const StreamModelPredictionsFacemesh_js_1 = require("./StreamModelPredictionsFacemesh.js");
const StreamModelPredictionsJobDetails_js_1 = require("./StreamModelPredictionsJobDetails.js");
const StreamModelPredictionsLanguage_js_1 = require("./StreamModelPredictionsLanguage.js");
const StreamModelPredictionsProsody_js_1 = require("./StreamModelPredictionsProsody.js");
exports.StreamModelPredictions = core.serialization.object({
    payloadId: core.serialization.property("payload_id", core.serialization.string().optional()),
    jobDetails: core.serialization.property("job_details", StreamModelPredictionsJobDetails_js_1.StreamModelPredictionsJobDetails.optional()),
    burst: StreamModelPredictionsBurst_js_1.StreamModelPredictionsBurst.optional(),
    face: StreamModelPredictionsFace_js_1.StreamModelPredictionsFace.optional(),
    facemesh: StreamModelPredictionsFacemesh_js_1.StreamModelPredictionsFacemesh.optional(),
    language: StreamModelPredictionsLanguage_js_1.StreamModelPredictionsLanguage.optional(),
    prosody: StreamModelPredictionsProsody_js_1.StreamModelPredictionsProsody.optional(),
});
