// hume/dist/cjs/api/resources/tts/resources/streamInput/client/Client.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamInputClient = void 0;
const BaseClient_js_1 = require("../../../../../../BaseClient.js");
const headers_js_1 = require("../../../../../../core/headers.js");
const core = __importStar(require("../../../../../../core/index.js"));
const environments = __importStar(require("../../../../../../environments.js"));
const serializers = __importStar(require("../../../../../../serialization/index.js"));
const Socket_js_1 = require("./Socket.js");
class StreamInputClient {
    constructor(options = {}) {
        this._options = (0, BaseClient_js_1.normalizeClientOptions)(options);
    }
    connect() {
        return __awaiter(this, arguments, void 0, function* (args = {}) {
            var _a, _b;
            const { accessToken, contextGenerationId, formatType, includeTimestampTypes, instantMode, noBinary, stripHeaders, version, apiKey, headers, debug, reconnectAttempts, } = args;
            const _queryParams = {};
            if (accessToken != null) {
                _queryParams.access_token = accessToken;
            }
            if (contextGenerationId != null) {
                _queryParams.context_generation_id = contextGenerationId;
            }
            if (formatType != null) {
                _queryParams.format_type = serializers.tts.AudioFormatType.jsonOrThrow(formatType, {
                    unrecognizedObjectKeys: "strip",
                    omitUndefined: true,
                });
            }
            if (includeTimestampTypes != null) {
                if (Array.isArray(includeTimestampTypes)) {
                    _queryParams.include_timestamp_types = includeTimestampTypes.map((item) => serializers.tts.TimestampType.jsonOrThrow(item, {
                        unrecognizedObjectKeys: "strip",
                        omitUndefined: true,
                    }));
                }
                else {
                    _queryParams.include_timestamp_types = serializers.tts.TimestampType.jsonOrThrow(includeTimestampTypes, {
                        unrecognizedObjectKeys: "strip",
                        omitUndefined: true,
                    });
                }
            }
            if (instantMode != null) {
                _queryParams.instant_mode = instantMode.toString();
            }
            if (noBinary != null) {
                _queryParams.no_binary = noBinary.toString();
            }
            if (stripHeaders != null) {
                _queryParams.strip_headers = stripHeaders.toString();
            }
            if (version != null) {
                _queryParams.version = serializers.tts.OctaveVersion.jsonOrThrow(version, {
                    unrecognizedObjectKeys: "strip",
                    omitUndefined: true,
                });
            }
            if (apiKey != null) {
                _queryParams.api_key = apiKey;
            }
            const _headers = (0, headers_js_1.mergeHeaders)((0, headers_js_1.mergeOnlyDefinedHeaders)(Object.assign({}, (yield this._getCustomAuthorizationHeaders()))), headers);
            const socket = new core.ReconnectingWebSocket({
                url: core.url.join((_a = (yield core.Supplier.get(this._options.baseUrl))) !== null && _a !== void 0 ? _a : ((_b = (yield core.Supplier.get(this._options.environment))) !== null && _b !== void 0 ? _b : environments.HumeEnvironment.Prod).tts, "/stream/input"),
                protocols: [],
                queryParameters: _queryParams,
                headers: _headers,
                options: { debug: debug !== null && debug !== void 0 ? debug : false, maxRetries: reconnectAttempts !== null && reconnectAttempts !== void 0 ? reconnectAttempts : 30 },
            });
            return new Socket_js_1.StreamInputSocket({ socket });
        });
    }
    _getCustomAuthorizationHeaders() {
        return __awaiter(this, void 0, void 0, function* () {
            var _a;
            const apiKeyValue = core.Supplier.get(this._options.apiKey);
            // This `authHeaderValue` is manually added as if you don't provide it it will
            // be omitted from the headers which means it won't reach the logic in ws.ts that
            // extracts values from the headers and adds them to query parameters.
            const authHeaderValue = core.Supplier.get((_a = this._options.headers) === null || _a === void 0 ? void 0 : _a.authorization);
            return { "X-Hume-Api-Key": apiKeyValue, Authorization: authHeaderValue };
        });
    }
}
exports.StreamInputClient = StreamInputClient;
