// hume/dist/esm/serialization/resources/expressionMeasurement/resources/stream/resources/stream/types/StreamModelPredictions.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../../../../../core/index.mjs";
import { StreamModelPredictionsBurst } from "./StreamModelPredictionsBurst.mjs";
import { StreamModelPredictionsFace } from "./StreamModelPredictionsFace.mjs";
import { StreamModelPredictionsFacemesh } from "./StreamModelPredictionsFacemesh.mjs";
import { StreamModelPredictionsJobDetails } from "./StreamModelPredictionsJobDetails.mjs";
import { StreamModelPredictionsLanguage } from "./StreamModelPredictionsLanguage.mjs";
import { StreamModelPredictionsProsody } from "./StreamModelPredictionsProsody.mjs";
export const StreamModelPredictions = core.serialization.object({
    payloadId: core.serialization.property("payload_id", core.serialization.string().optional()),
    jobDetails: core.serialization.property("job_details", StreamModelPredictionsJobDetails.optional()),
    burst: StreamModelPredictionsBurst.optional(),
    face: StreamModelPredictionsFace.optional(),
    facemesh: StreamModelPredictionsFacemesh.optional(),
    language: StreamModelPredictionsLanguage.optional(),
    prosody: StreamModelPredictionsProsody.optional(),
});
