// hume/dist/esm/serialization/resources/empathicVoice/resources/configs/client/requests/PostedConfigVersion.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../../../../core/index.mjs";
import { PostedBuiltinTool } from "../../../../types/PostedBuiltinTool.mjs";
import { PostedConfigPromptSpec } from "../../../../types/PostedConfigPromptSpec.mjs";
import { PostedEllmModel } from "../../../../types/PostedEllmModel.mjs";
import { PostedEventMessageSpecs } from "../../../../types/PostedEventMessageSpecs.mjs";
import { PostedLanguageModel } from "../../../../types/PostedLanguageModel.mjs";
import { PostedNudgeSpec } from "../../../../types/PostedNudgeSpec.mjs";
import { PostedTimeoutSpecs } from "../../../../types/PostedTimeoutSpecs.mjs";
import { PostedUserDefinedToolSpec } from "../../../../types/PostedUserDefinedToolSpec.mjs";
import { PostedWebhookSpec } from "../../../../types/PostedWebhookSpec.mjs";
import { VoiceRef } from "../../../../types/VoiceRef.mjs";
export const PostedConfigVersion = core.serialization.object({
    builtinTools: core.serialization.property("builtin_tools", core.serialization.list(PostedBuiltinTool.optional()).optional()),
    ellmModel: core.serialization.property("ellm_model", PostedEllmModel.optional()),
    eventMessages: core.serialization.property("event_messages", PostedEventMessageSpecs.optional()),
    eviVersion: core.serialization.property("evi_version", core.serialization.string()),
    languageModel: core.serialization.property("language_model", PostedLanguageModel.optional()),
    nudges: PostedNudgeSpec.optional(),
    prompt: PostedConfigPromptSpec.optional(),
    timeouts: PostedTimeoutSpecs.optional(),
    tools: core.serialization.list(PostedUserDefinedToolSpec.optional()).optional(),
    versionDescription: core.serialization.property("version_description", core.serialization.string().optional()),
    voice: VoiceRef.optional(),
    webhooks: core.serialization.list(PostedWebhookSpec.optional()).optional(),
});
