// hume/dist/esm/api/resources/expressionMeasurement/resources/stream/resources/stream/client/Client.mjs
// This file was auto-generated by Fern from our API Definition.
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { normalizeClientOptions } from "../../../../../../../../BaseClient.mjs";
import { mergeHeaders, mergeOnlyDefinedHeaders } from "../../../../../../../../core/headers.mjs";
import * as core from "../../../../../../../../core/index.mjs";
import * as environments from "../../../../../../../../environments.mjs";
import { StreamSocket } from "./Socket.mjs";
export class StreamClient {
    constructor(options = {}) {
        this._options = normalizeClientOptions(options);
    }
    connect(args) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a, _b;
            const { queryParams, headers, debug, reconnectAttempts } = args;
            const _headers = mergeHeaders(mergeOnlyDefinedHeaders({ "X-Hume-Api-Key": args["X-Hume-Api-Key"] }), headers);
            const socket = new core.ReconnectingWebSocket({
                url: core.url.join((_a = (yield core.Supplier.get(this._options.baseUrl))) !== null && _a !== void 0 ? _a : ((_b = (yield core.Supplier.get(this._options.environment))) !== null && _b !== void 0 ? _b : environments.HumeEnvironment.Prod).stream, "/models"),
                protocols: [],
                queryParameters: (queryParams !== null && queryParams !== void 0 ? queryParams : {}),
                headers: _headers,
                options: { debug: debug !== null && debug !== void 0 ? debug : false, maxRetries: reconnectAttempts !== null && reconnectAttempts !== void 0 ? reconnectAttempts : 30 },
            });
            return new StreamSocket({ socket });
        });
    }
}
