// hume/dist/esm/auth/HeaderAuthProvider.mjs
/**
 * THIS FILE IS MANUALLY MAINTAINED: see .fernignore
 *
 * Custom HeaderAuthProvider that supports both apiKey and accessToken authentication.
 * When accessToken is provided to HumeClient, the wrapper converts it to an
 * Authorization: Bearer header. This provider checks for both apiKey and the
 * Authorization header to support both auth methods.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import * as core from "../core/index.mjs";
import * as errors from "../errors/index.mjs";
export class HeaderAuthProvider {
    constructor(options) {
        this.options = options;
    }
    static canCreate(options) {
        return options.apiKey != null || HeaderAuthProvider.hasAuthorizationHeader(options.headers);
    }
    static hasAuthorizationHeader(headers) {
        if (!headers)
            return false;
        return Object.keys(headers).some((key) => key.toLowerCase() === "authorization");
    }
    getAuthRequest(_arg) {
        return __awaiter(this, void 0, void 0, function* () {
            // Check apiKey first (preferred method)
            const apiKey = yield core.Supplier.get(this.options.apiKey);
            if (apiKey != null) {
                return { headers: { "X-Hume-Api-Key": apiKey } };
            }
            // Check for Authorization header (from accessToken)
            if (this.options.headers) {
                const authKey = Object.keys(this.options.headers).find((key) => key.toLowerCase() === "authorization");
                if (authKey) {
                    const authValue = yield core.Supplier.get(this.options.headers[authKey]);
                    if (authValue != null) {
                        return { headers: { Authorization: authValue } };
                    }
                }
            }
            throw new errors.HumeError({
                message: "Please provide either 'apiKey' or 'accessToken' when initializing the client",
            });
        });
    }
}
