// hume/dist/cjs/auth/HeaderAuthProvider.js
"use strict";
/**
 * THIS FILE IS MANUALLY MAINTAINED: see .fernignore
 *
 * Custom HeaderAuthProvider that supports both apiKey and accessToken authentication.
 * When accessToken is provided to HumeClient, the wrapper converts it to an
 * Authorization: Bearer header. This provider checks for both apiKey and the
 * Authorization header to support both auth methods.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HeaderAuthProvider = void 0;
const core = __importStar(require("../core/index.js"));
const errors = __importStar(require("../errors/index.js"));
class HeaderAuthProvider {
    constructor(options) {
        this.options = options;
    }
    static canCreate(options) {
        return options.apiKey != null || HeaderAuthProvider.hasAuthorizationHeader(options.headers);
    }
    static hasAuthorizationHeader(headers) {
        if (!headers)
            return false;
        return Object.keys(headers).some((key) => key.toLowerCase() === "authorization");
    }
    getAuthRequest(_arg) {
        return __awaiter(this, void 0, void 0, function* () {
            // Check apiKey first (preferred method)
            const apiKey = yield core.Supplier.get(this.options.apiKey);
            if (apiKey != null) {
                return { headers: { "X-Hume-Api-Key": apiKey } };
            }
            // Check for Authorization header (from accessToken)
            if (this.options.headers) {
                const authKey = Object.keys(this.options.headers).find((key) => key.toLowerCase() === "authorization");
                if (authKey) {
                    const authValue = yield core.Supplier.get(this.options.headers[authKey]);
                    if (authValue != null) {
                        return { headers: { Authorization: authValue } };
                    }
                }
            }
            throw new errors.HumeError({
                message: "Please provide either 'apiKey' or 'accessToken' when initializing the client",
            });
        });
    }
}
exports.HeaderAuthProvider = HeaderAuthProvider;
