// hume/dist/esm/serialization/resources/expressionMeasurement/resources/batch/types/ModelsPredictions.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../../../core/index.mjs";
import { PredictionsOptionalNullBurstPrediction } from "./PredictionsOptionalNullBurstPrediction.mjs";
import { PredictionsOptionalNullFacemeshPrediction } from "./PredictionsOptionalNullFacemeshPrediction.mjs";
import { PredictionsOptionalNullFacePrediction } from "./PredictionsOptionalNullFacePrediction.mjs";
import { PredictionsOptionalTranscriptionMetadataLanguagePrediction } from "./PredictionsOptionalTranscriptionMetadataLanguagePrediction.mjs";
import { PredictionsOptionalTranscriptionMetadataNerPrediction } from "./PredictionsOptionalTranscriptionMetadataNerPrediction.mjs";
import { PredictionsOptionalTranscriptionMetadataProsodyPrediction } from "./PredictionsOptionalTranscriptionMetadataProsodyPrediction.mjs";
export const ModelsPredictions = core.serialization.object({
    face: PredictionsOptionalNullFacePrediction.optional(),
    burst: PredictionsOptionalNullBurstPrediction.optional(),
    prosody: PredictionsOptionalTranscriptionMetadataProsodyPrediction.optional(),
    language: PredictionsOptionalTranscriptionMetadataLanguagePrediction.optional(),
    ner: PredictionsOptionalTranscriptionMetadataNerPrediction.optional(),
    facemesh: PredictionsOptionalNullFacemeshPrediction.optional(),
});
