// hume/dist/esm/serialization/resources/empathicVoice/types/ConnectSessionSettings.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { ConnectSessionSettingsAudio } from "./ConnectSessionSettingsAudio.mjs";
import { ConnectSessionSettingsContext } from "./ConnectSessionSettingsContext.mjs";
import { ConnectSessionSettingsVariablesValue } from "./ConnectSessionSettingsVariablesValue.mjs";
export const ConnectSessionSettings = core.serialization.object({
    audio: ConnectSessionSettingsAudio.optional(),
    context: ConnectSessionSettingsContext.optional(),
    customSessionId: core.serialization.property("custom_session_id", core.serialization.string().optional()),
    eventLimit: core.serialization.property("event_limit", core.serialization.number().optional()),
    languageModelApiKey: core.serialization.property("language_model_api_key", core.serialization.string().optional()),
    systemPrompt: core.serialization.property("system_prompt", core.serialization.string().optional()),
    voiceId: core.serialization.property("voice_id", core.serialization.string().optional()),
    variables: core.serialization.record(core.serialization.string(), ConnectSessionSettingsVariablesValue).optional(),
});
