// hume/dist/esm/serialization/resources/empathicVoice/types/ReturnChatPagedEvents.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { ReturnChatEvent } from "./ReturnChatEvent.mjs";
import { ReturnChatPagedEventsPaginationDirection } from "./ReturnChatPagedEventsPaginationDirection.mjs";
import { ReturnChatPagedEventsStatus } from "./ReturnChatPagedEventsStatus.mjs";
import { ReturnConfigSpec } from "./ReturnConfigSpec.mjs";
export const ReturnChatPagedEvents = core.serialization.object({
    chatGroupId: core.serialization.property("chat_group_id", core.serialization.string()),
    config: ReturnConfigSpec.optional(),
    endTimestamp: core.serialization.property("end_timestamp", core.serialization.number().optional()),
    eventsPage: core.serialization.property("events_page", core.serialization.list(ReturnChatEvent)),
    id: core.serialization.string(),
    metadata: core.serialization.string().optional(),
    pageNumber: core.serialization.property("page_number", core.serialization.number()),
    pageSize: core.serialization.property("page_size", core.serialization.number()),
    paginationDirection: core.serialization.property("pagination_direction", ReturnChatPagedEventsPaginationDirection),
    startTimestamp: core.serialization.property("start_timestamp", core.serialization.number()),
    status: ReturnChatPagedEventsStatus,
    totalPages: core.serialization.property("total_pages", core.serialization.number()),
});
