// hume/dist/esm/serialization/resources/empathicVoice/types/ReturnChatGroupPagedChats.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { ReturnChat } from "./ReturnChat.mjs";
import { ReturnChatGroupPagedChatsPaginationDirection } from "./ReturnChatGroupPagedChatsPaginationDirection.mjs";
export const ReturnChatGroupPagedChats = core.serialization.object({
    active: core.serialization.boolean().optional(),
    chatsPage: core.serialization.property("chats_page", core.serialization.list(ReturnChat)),
    firstStartTimestamp: core.serialization.property("first_start_timestamp", core.serialization.number()),
    id: core.serialization.string(),
    mostRecentStartTimestamp: core.serialization.property("most_recent_start_timestamp", core.serialization.number()),
    numChats: core.serialization.property("num_chats", core.serialization.number()),
    pageNumber: core.serialization.property("page_number", core.serialization.number()),
    pageSize: core.serialization.property("page_size", core.serialization.number()),
    paginationDirection: core.serialization.property("pagination_direction", ReturnChatGroupPagedChatsPaginationDirection),
    totalPages: core.serialization.property("total_pages", core.serialization.number()),
});
