// hume/dist/esm/serialization/resources/empathicVoice/types/ReturnChatGroupPagedAudioReconstructions.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { ReturnChatAudioReconstruction } from "./ReturnChatAudioReconstruction.mjs";
import { ReturnChatGroupPagedAudioReconstructionsPaginationDirection } from "./ReturnChatGroupPagedAudioReconstructionsPaginationDirection.mjs";
export const ReturnChatGroupPagedAudioReconstructions = core.serialization.object({
    audioReconstructionsPage: core.serialization.property("audio_reconstructions_page", core.serialization.list(ReturnChatAudioReconstruction)),
    id: core.serialization.string(),
    numChats: core.serialization.property("num_chats", core.serialization.number()),
    pageNumber: core.serialization.property("page_number", core.serialization.number()),
    pageSize: core.serialization.property("page_size", core.serialization.number()),
    paginationDirection: core.serialization.property("pagination_direction", ReturnChatGroupPagedAudioReconstructionsPaginationDirection),
    totalPages: core.serialization.property("total_pages", core.serialization.number()),
    userId: core.serialization.property("user_id", core.serialization.string()),
});
