// hume/dist/esm/serialization/resources/empathicVoice/types/ReturnUserDefinedTool.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { ReturnUserDefinedToolToolType } from "./ReturnUserDefinedToolToolType.mjs";
import { ReturnUserDefinedToolVersionType } from "./ReturnUserDefinedToolVersionType.mjs";
export const ReturnUserDefinedTool = core.serialization.object({
    createdOn: core.serialization.property("created_on", core.serialization.number()),
    description: core.serialization.string().optional(),
    fallbackContent: core.serialization.property("fallback_content", core.serialization.string().optional()),
    id: core.serialization.string(),
    modifiedOn: core.serialization.property("modified_on", core.serialization.number()),
    name: core.serialization.string(),
    parameters: core.serialization.string(),
    toolType: core.serialization.property("tool_type", ReturnUserDefinedToolToolType),
    version: core.serialization.number(),
    versionDescription: core.serialization.property("version_description", core.serialization.string().optional()),
    versionType: core.serialization.property("version_type", ReturnUserDefinedToolVersionType),
});
