// hume/dist/esm/serialization/resources/empathicVoice/types/ReturnConfig.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { ReturnBuiltinTool } from "./ReturnBuiltinTool.mjs";
import { ReturnEllmModel } from "./ReturnEllmModel.mjs";
import { ReturnEventMessageSpecs } from "./ReturnEventMessageSpecs.mjs";
import { ReturnLanguageModel } from "./ReturnLanguageModel.mjs";
import { ReturnNudgeSpec } from "./ReturnNudgeSpec.mjs";
import { ReturnPrompt } from "./ReturnPrompt.mjs";
import { ReturnTimeoutSpecs } from "./ReturnTimeoutSpecs.mjs";
import { ReturnUserDefinedTool } from "./ReturnUserDefinedTool.mjs";
import { ReturnVoice } from "./ReturnVoice.mjs";
import { ReturnWebhookSpec } from "./ReturnWebhookSpec.mjs";
export const ReturnConfig = core.serialization.object({
    builtinTools: core.serialization.property("builtin_tools", core.serialization.list(ReturnBuiltinTool.optional()).optional()),
    createdOn: core.serialization.property("created_on", core.serialization.number().optional()),
    ellmModel: core.serialization.property("ellm_model", ReturnEllmModel.optional()),
    eventMessages: core.serialization.property("event_messages", ReturnEventMessageSpecs.optional()),
    eviVersion: core.serialization.property("evi_version", core.serialization.string().optional()),
    id: core.serialization.string().optional(),
    languageModel: core.serialization.property("language_model", ReturnLanguageModel.optional()),
    modifiedOn: core.serialization.property("modified_on", core.serialization.number().optional()),
    name: core.serialization.string().optional(),
    nudges: ReturnNudgeSpec.optional(),
    prompt: ReturnPrompt.optional(),
    timeouts: ReturnTimeoutSpecs.optional(),
    tools: core.serialization.list(ReturnUserDefinedTool.optional()).optional(),
    version: core.serialization.number().optional(),
    versionDescription: core.serialization.property("version_description", core.serialization.string().optional()),
    voice: ReturnVoice.optional(),
    webhooks: core.serialization.list(ReturnWebhookSpec.optional()).optional(),
});
