// hume/dist/esm/serialization/resources/empathicVoice/types/SessionSettings.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { AudioConfiguration } from "./AudioConfiguration.mjs";
import { BuiltinToolConfig } from "./BuiltinToolConfig.mjs";
import { Context } from "./Context.mjs";
import { SessionSettingsVariablesValue } from "./SessionSettingsVariablesValue.mjs";
import { Tool } from "./Tool.mjs";
export const SessionSettings = core.serialization.object({
    audio: AudioConfiguration.optional(),
    builtinTools: core.serialization.property("builtin_tools", core.serialization.list(BuiltinToolConfig).optional()),
    context: Context.optional(),
    customSessionId: core.serialization.property("custom_session_id", core.serialization.string().optional()),
    languageModelApiKey: core.serialization.property("language_model_api_key", core.serialization.string().optional()),
    metadata: core.serialization.record(core.serialization.string(), core.serialization.unknown()).optional(),
    systemPrompt: core.serialization.property("system_prompt", core.serialization.string().optional()),
    tools: core.serialization.list(Tool).optional(),
    type: core.serialization.stringLiteral("session_settings"),
    variables: core.serialization.record(core.serialization.string(), SessionSettingsVariablesValue).optional(),
    voiceId: core.serialization.property("voice_id", core.serialization.string().optional()),
});
