// hume/dist/esm/serialization/resources/empathicVoice/types/WebhookEventChatEnded.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { WebhookEventBase } from "./WebhookEventBase.mjs";
import { WebhookEventChatStatus } from "./WebhookEventChatStatus.mjs";
export const WebhookEventChatEnded = core.serialization
    .object({
    callerNumber: core.serialization.property("caller_number", core.serialization.string().optional()),
    customSessionId: core.serialization.property("custom_session_id", core.serialization.string().optional()),
    durationSeconds: core.serialization.property("duration_seconds", core.serialization.number()),
    endReason: core.serialization.property("end_reason", WebhookEventChatStatus),
    endTime: core.serialization.property("end_time", core.serialization.number()),
    eventName: core.serialization.property("event_name", core.serialization.stringLiteral("chat_ended").optional()),
    twilioMetadata: core.serialization.property("twilio_metadata", core.serialization.record(core.serialization.string(), core.serialization.string().optional()).optional()),
})
    .extend(WebhookEventBase);
