// hume/dist/esm/serialization/resources/empathicVoice/types/WebhookEventChatStarted.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { WebhookEventBase } from "./WebhookEventBase.mjs";
import { WebhookEventChatStartType } from "./WebhookEventChatStartType.mjs";
export const WebhookEventChatStarted = core.serialization
    .object({
    callerNumber: core.serialization.property("caller_number", core.serialization.string().optional()),
    chatStartType: core.serialization.property("chat_start_type", WebhookEventChatStartType),
    customSessionId: core.serialization.property("custom_session_id", core.serialization.string().optional()),
    eventName: core.serialization.property("event_name", core.serialization.stringLiteral("chat_started").optional()),
    startTime: core.serialization.property("start_time", core.serialization.number()),
    twilioMetadata: core.serialization.property("twilio_metadata", core.serialization.record(core.serialization.string(), core.serialization.string().optional()).optional()),
})
    .extend(WebhookEventBase);
