// hume/dist/cjs/serialization/resources/empathicVoice/types/ReturnConfig.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReturnConfig = void 0;
const core = __importStar(require("../../../../core/index.js"));
const ReturnBuiltinTool_js_1 = require("./ReturnBuiltinTool.js");
const ReturnEllmModel_js_1 = require("./ReturnEllmModel.js");
const ReturnEventMessageSpecs_js_1 = require("./ReturnEventMessageSpecs.js");
const ReturnLanguageModel_js_1 = require("./ReturnLanguageModel.js");
const ReturnNudgeSpec_js_1 = require("./ReturnNudgeSpec.js");
const ReturnPrompt_js_1 = require("./ReturnPrompt.js");
const ReturnTimeoutSpecs_js_1 = require("./ReturnTimeoutSpecs.js");
const ReturnUserDefinedTool_js_1 = require("./ReturnUserDefinedTool.js");
const ReturnVoice_js_1 = require("./ReturnVoice.js");
const ReturnWebhookSpec_js_1 = require("./ReturnWebhookSpec.js");
exports.ReturnConfig = core.serialization.object({
    builtinTools: core.serialization.property("builtin_tools", core.serialization.list(ReturnBuiltinTool_js_1.ReturnBuiltinTool.optional()).optional()),
    createdOn: core.serialization.property("created_on", core.serialization.number().optional()),
    ellmModel: core.serialization.property("ellm_model", ReturnEllmModel_js_1.ReturnEllmModel.optional()),
    eventMessages: core.serialization.property("event_messages", ReturnEventMessageSpecs_js_1.ReturnEventMessageSpecs.optional()),
    eviVersion: core.serialization.property("evi_version", core.serialization.string().optional()),
    id: core.serialization.string().optional(),
    languageModel: core.serialization.property("language_model", ReturnLanguageModel_js_1.ReturnLanguageModel.optional()),
    modifiedOn: core.serialization.property("modified_on", core.serialization.number().optional()),
    name: core.serialization.string().optional(),
    nudges: ReturnNudgeSpec_js_1.ReturnNudgeSpec.optional(),
    prompt: ReturnPrompt_js_1.ReturnPrompt.optional(),
    timeouts: ReturnTimeoutSpecs_js_1.ReturnTimeoutSpecs.optional(),
    tools: core.serialization.list(ReturnUserDefinedTool_js_1.ReturnUserDefinedTool.optional()).optional(),
    version: core.serialization.number().optional(),
    versionDescription: core.serialization.property("version_description", core.serialization.string().optional()),
    voice: ReturnVoice_js_1.ReturnVoice.optional(),
    webhooks: core.serialization.list(ReturnWebhookSpec_js_1.ReturnWebhookSpec.optional()).optional(),
});
