// hume/dist/cjs/serialization/resources/empathicVoice/types/ReturnChat.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReturnChat = void 0;
const core = __importStar(require("../../../../core/index.js"));
const ReturnChatStatus_js_1 = require("./ReturnChatStatus.js");
const ReturnConfigSpec_js_1 = require("./ReturnConfigSpec.js");
exports.ReturnChat = core.serialization.object({
    chatGroupId: core.serialization.property("chat_group_id", core.serialization.string()),
    config: ReturnConfigSpec_js_1.ReturnConfigSpec.optional(),
    endTimestamp: core.serialization.property("end_timestamp", core.serialization.number().optional()),
    eventCount: core.serialization.property("event_count", core.serialization.number().optional()),
    id: core.serialization.string(),
    metadata: core.serialization.string().optional(),
    startTimestamp: core.serialization.property("start_timestamp", core.serialization.number()),
    status: ReturnChatStatus_js_1.ReturnChatStatus,
});
