// hume/dist/cjs/serialization/resources/empathicVoice/types/WebhookEventChatStarted.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookEventChatStarted = void 0;
const core = __importStar(require("../../../../core/index.js"));
const WebhookEventBase_js_1 = require("./WebhookEventBase.js");
const WebhookEventChatStartType_js_1 = require("./WebhookEventChatStartType.js");
exports.WebhookEventChatStarted = core.serialization
    .object({
    callerNumber: core.serialization.property("caller_number", core.serialization.string().optional()),
    chatStartType: core.serialization.property("chat_start_type", WebhookEventChatStartType_js_1.WebhookEventChatStartType),
    customSessionId: core.serialization.property("custom_session_id", core.serialization.string().optional()),
    eventName: core.serialization.property("event_name", core.serialization.stringLiteral("chat_started").optional()),
    startTime: core.serialization.property("start_time", core.serialization.number()),
    twilioMetadata: core.serialization.property("twilio_metadata", core.serialization.record(core.serialization.string(), core.serialization.string().optional()).optional()),
})
    .extend(WebhookEventBase_js_1.WebhookEventBase);
