// hume/dist/cjs/serialization/resources/expressionMeasurement/resources/batch/types/ModelsPredictions.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelsPredictions = void 0;
const core = __importStar(require("../../../../../../core/index.js"));
const PredictionsOptionalNullBurstPrediction_js_1 = require("./PredictionsOptionalNullBurstPrediction.js");
const PredictionsOptionalNullFacemeshPrediction_js_1 = require("./PredictionsOptionalNullFacemeshPrediction.js");
const PredictionsOptionalNullFacePrediction_js_1 = require("./PredictionsOptionalNullFacePrediction.js");
const PredictionsOptionalTranscriptionMetadataLanguagePrediction_js_1 = require("./PredictionsOptionalTranscriptionMetadataLanguagePrediction.js");
const PredictionsOptionalTranscriptionMetadataNerPrediction_js_1 = require("./PredictionsOptionalTranscriptionMetadataNerPrediction.js");
const PredictionsOptionalTranscriptionMetadataProsodyPrediction_js_1 = require("./PredictionsOptionalTranscriptionMetadataProsodyPrediction.js");
exports.ModelsPredictions = core.serialization.object({
    face: PredictionsOptionalNullFacePrediction_js_1.PredictionsOptionalNullFacePrediction.optional(),
    burst: PredictionsOptionalNullBurstPrediction_js_1.PredictionsOptionalNullBurstPrediction.optional(),
    prosody: PredictionsOptionalTranscriptionMetadataProsodyPrediction_js_1.PredictionsOptionalTranscriptionMetadataProsodyPrediction.optional(),
    language: PredictionsOptionalTranscriptionMetadataLanguagePrediction_js_1.PredictionsOptionalTranscriptionMetadataLanguagePrediction.optional(),
    ner: PredictionsOptionalTranscriptionMetadataNerPrediction_js_1.PredictionsOptionalTranscriptionMetadataNerPrediction.optional(),
    facemesh: PredictionsOptionalNullFacemeshPrediction_js_1.PredictionsOptionalNullFacemeshPrediction.optional(),
});
