// hume/dist/cjs/serialization/resources/expressionMeasurement/resources/batch/types/StateTraining.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.StateTraining = void 0;
const core = __importStar(require("../../../../../../core/index.js"));
const StateTrainingCompletedTraining_js_1 = require("./StateTrainingCompletedTraining.js");
const StateTrainingFailed_js_1 = require("./StateTrainingFailed.js");
const StateTrainingInProgress_js_1 = require("./StateTrainingInProgress.js");
const StateTrainingQueued_js_1 = require("./StateTrainingQueued.js");
exports.StateTraining = core.serialization
    .union("status", {
    QUEUED: StateTrainingQueued_js_1.StateTrainingQueued,
    IN_PROGRESS: StateTrainingInProgress_js_1.StateTrainingInProgress,
    COMPLETED: StateTrainingCompletedTraining_js_1.StateTrainingCompletedTraining,
    FAILED: StateTrainingFailed_js_1.StateTrainingFailed,
})
    .transform({
    transform: (value) => value,
    untransform: (value) => value,
});
