// hume/dist/esm/serialization/resources/tts/types/PostedTts.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { Format } from "./Format.mjs";
import { OctaveVersion } from "./OctaveVersion.mjs";
import { PostedContext } from "./PostedContext.mjs";
import { PostedUtterance } from "./PostedUtterance.mjs";
import { TimestampType } from "./TimestampType.mjs";
export const PostedTts = core.serialization.object({
    context: PostedContext.optional(),
    format: Format.optional(),
    includeTimestampTypes: core.serialization.property("include_timestamp_types", core.serialization.list(TimestampType).optional()),
    numGenerations: core.serialization.property("num_generations", core.serialization.number().optional()),
    splitUtterances: core.serialization.property("split_utterances", core.serialization.boolean().optional()),
    stripHeaders: core.serialization.property("strip_headers", core.serialization.boolean().optional()),
    utterances: core.serialization.list(PostedUtterance),
    version: OctaveVersion.optional(),
    instantMode: core.serialization.property("instant_mode", core.serialization.boolean().optional()),
});
