// hume/dist/esm/serialization/resources/tts/types/SnippetAudioChunk.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { AudioFormatType } from "./AudioFormatType.mjs";
import { Snippet } from "./Snippet.mjs";
export const SnippetAudioChunk = core.serialization.object({
    audio: core.serialization.string(),
    audioFormat: core.serialization.property("audio_format", AudioFormatType),
    chunkIndex: core.serialization.property("chunk_index", core.serialization.number()),
    generationId: core.serialization.property("generation_id", core.serialization.string()),
    isLastChunk: core.serialization.property("is_last_chunk", core.serialization.boolean()),
    requestId: core.serialization.property("request_id", core.serialization.string()),
    snippet: Snippet.optional(),
    snippetId: core.serialization.property("snippet_id", core.serialization.string()),
    text: core.serialization.string(),
    transcribedText: core.serialization.property("transcribed_text", core.serialization.string().optional()),
    type: core.serialization.stringLiteral("audio"),
    utteranceIndex: core.serialization.property("utterance_index", core.serialization.number().optional()),
});
